src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -228,16 +239,19 @@ public void publish(MqttMessage message) throws MessageClientException {
     public void updateSubscriptions(Set<String> topics, Consumer<MqttMessage> messageHandler) {
         this.messageHandler = messageHandler;
 
-        this.toSubscribeLocalMqttTopics = new HashSet<>(topics);
+        synchronized (subscribeLock) {
+            toSubscribeLocalMqttTopics.clear();
+            toSubscribeLocalMqttTopics.addAll(topics);
+        }
         LOGGER.atDebug().kv("topics", topics).log("Updated local MQTT topics to subscribe");
 
         if (mqttClientInternal.isConnected()) {
-            updateSubscriptionsInternal();
+            updateSubscriptionsAsync();
         }
     }
 
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    private void updateSubscriptionsInternal() {
+    private void updateSubscriptionsAsync() {
         synchronized (subscribeLock) {
```

Comment:
this doesn't look async. We're unsubscribing from all topics synchronously. This method also then doesn't subscribe to any topics?